### PR TITLE
TR PIM 4936 - Fixes on import performances and memory leak

### PR DESCRIPTION
**Pull request by BitOne**
| Q                  | A
| ------------------ | ---
| Migration script?  | N.A.
| Behats             | Running
| Specs & Static     | OK
| Changelog updated  | OK

### CODE REVIEW

**Comment by jjanvier** [comment]
GTM

File: src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php [hunk 1]
```diff
@@ -139,6 +143,7 @@ protected function createVersion($versionable)
         $versions = $this->versionManager->buildVersion($versionable, $changeset);
 
         foreach ($versions as $version) {
+            $this->versions[] = $version;
```
Comment (grena) [inline src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php]: I don't get why this array reset is in a foreach

**Comment by grena** [comment]
GTM

**Comment by solivier** [comment]
GTM

File: src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php [hunk 2]
```diff
@@ -226,4 +231,18 @@ protected function getObjectHash($object)
     {
         return sprintf('%s#%s', spl_object_hash($object), sha1($this->versionContext->getContextInfo()));
     }
+
+    /**
+     * Clear versions know to this subscribler from the object manager
```
Comment (solivier) [inline src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php]: subscriber

File: CHANGELOG-1.4.md [hunk 3]
```diff
@@ -6,7 +6,8 @@
 - PIM-4887: Fixed locales active status when removed from channels
 - PIM-4911: Fix escaping of property with locale and scope
 - PIM-4922: Fix media attribute preview
-- PIM-4925: Fix dashboard patch information 
+- PIM-4925: Fix dashboard patch information
+- PIM-4936: Fixes performances problems and memory leak at import time
```
Comment (phaseinducer) [inline CHANGELOG-1.4.md]: No third person for homogeneity please

File: src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php [hunk 4]
```diff
@@ -29,6 +29,9 @@ class AddVersionSubscriber implements EventSubscriber
     /** @var string[] */
     protected $versionedEntities = [];
 
+    /** @var string[] */
```
Comment (solivier) [inline src/Pim/Bundle/VersioningBundle/EventSubscriber/AddVersionSubscriber.php]: string ?

**Pull request merged**